client/src/actions/annotation.js

```diff
@@ -325,15 +325,12 @@ export const saveObsAnnotationsAction = () => async (dispatch, getState) => {
   const state = getState();
   const { annotations, autosave } = state;
   const { dataCollectionNameIsReadOnly, dataCollectionName } = annotations;
-  const {
-    lastSavedAnnoMatrix,
-    saveInProgress,
-    exponentialBackoffFetch,
-  } = autosave;
+  const { lastSavedAnnoMatrix, exponentialBackoffFetch } = autosave;
```

Comment:
remove `saveInProgress` from `if (saveInProgress || annoMatrix === lastSavedAnnoMatrix) return;`, so `exponentialBackoffFetch()` gets called synchronously to allow either cancel timeout if `hash` has changed, or just return the resolved/rejected promise